### PR TITLE
LVUIDF-58: cover Article with tests

### PR DESCRIPTION
**Pull request by yurko**
https://ssu-jira.softserveinc.com/browse/LVUIDF-58

- [x] Cover Article model with tests
- [x] Cover Articles controller with tests

To review:
- [ ] @rtriska
- [x] @forecome 

Optional:
- [ ] @irynar

To merge:
- [ ] @izagor
- [x] @forecome

### CODE REVIEW

**Comment by yurko** [comment]
@forecome, please review

File: app/controllers/api/v1/articles_controller.rb [hunk 1]
```diff
@@ -33,22 +33,30 @@ def create
   end
 
   def update
-    if @article.update(article_params)
-      render json: @article
+    if set_article
```
Comment (forecome) [inline app/controllers/api/v1/articles_controller.rb]: `set_article` is before action method, it runs before `update`, you should not call it here. If you are trying to check that `@article` is assigned, than just change line 37 to `if @article && @article.update(article_params)`

File: app/controllers/api/v1/articles_controller.rb [hunk 2]
```diff
@@ -33,22 +33,30 @@ def create
...
     else
-      head :unprocessable_entity
+      head :forbidden
     end
   end
 
   def destroy
-    @article.destroy
-    head :no_content
+    if set_article
```
Comment (forecome) [inline app/controllers/api/v1/articles_controller.rb]: :arrow_up:

File: app/controllers/api/v1/articles_controller.rb [hunk 3]
```diff
@@ -33,22 +33,30 @@ def create
...
+      @article.destroy
+      head :no_content
+    else
+      head :not_found
```
Comment (forecome) [inline app/controllers/api/v1/articles_controller.rb]: I think you should response here with `forbidden` as well

File: app/controllers/api/v1/articles_controller.rb [hunk 4]
```diff
@@ -33,22 +33,30 @@ def create
...
+    end
   end
 
   private
 
   def set_article
-    @article = current_user.articles.find(params[:id])
+    @article = current_user.articles.find(params[:id]) if current_user
```
Comment (forecome) [inline app/controllers/api/v1/articles_controller.rb]: Look at first comment

File: app/controllers/api/v1/articles_controller.rb (hunk 1 above)
Comment (yurko) [inline app/controllers/api/v1/articles_controller.rb]: @forecome, fixed, please review

File: app/controllers/api/v1/articles_controller.rb [hunk 5]
```diff
@@ -41,14 +41,21 @@ def update
...
   end
 
   private
 
   def set_article
-    @article = current_user.articles.find(params[:id])
+    if current_user
+      @article = current_user.articles.find(params[:id])
+    else
+      head :forbidden
```
Comment (forecome) [inline app/controllers/api/v1/articles_controller.rb]: Please, make another before action for this. And remove `if @article` from action methods, as I wrote in previous comment.

File: spec/factories/users_factory.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :user do
+    sequence(:email){|n| "user#{n}@ls.com" }
```
Comment (forecome) [inline spec/factories/users_factory.rb]: Please, use `example.com` in any dummy email data.

File: app/controllers/api/v1/articles_controller.rb (hunk 5 above)
Comment (yurko) [inline app/controllers/api/v1/articles_controller.rb]: @forecome, reworked. please review

File: app/controllers/api/v1/articles_controller.rb [hunk 7]
```diff
@@ -41,12 +42,19 @@ def update
   end
 
   def destroy
-    @article.destroy
-    head :no_content
+    if @article.destroy
+      head :no_content
+    else
+      head :forbidden
```
Comment (forecome) [inline app/controllers/api/v1/articles_controller.rb]: Please, remove this, you have added before action

Comment (yurko) [inline app/controllers/api/v1/articles_controller.rb]: removed

**Pull request merged**